phoenix-ios/phoenix-ios/sync/SyncManagerState.swift

```diff
@@ -0,0 +1,308 @@
+import Foundation
+import CloudKit
+
+/* SyncManager State Machine:
+ *
+ * The state is always one of the following:
+ *
+ * - initializing
+ * - updatingCloud
+ *   - creatingRecordZone
+ *   - deletingRecordZone
+ * - downloading
+ * - uploading
+ * - waiting
+ *   - forCloudCredentials
+ *   - forInternet
+ *   - exponentialBackoff
+ *   - randomizedUploadDelay
+ * - synced
+ * - disabled
```

Comment:
So for example, if the wallet is downloading data from the cloud and we make a payment in the meantime, this new payment will not be backed up before the download completes?

Reply:
That's correct. That could be improved. I think we should leave that improvement for a future PR though.